src/library/commissioner_impl.cpp

```diff
@@ -1510,7 +1510,8 @@ Error CommissionerImpl::DecodeActiveOperationalDataset(ActiveOperationalDataset
         dataset.mPresentFlags |= ActiveOperationalDataset::kSecurityPolicyBit;
     }
 
-    aDataset = dataset;
+    dataset.mRawTlvs = aResponse.GetPayload();
```

Comment:
@simonlingoogle I think we should prefer getting the binary Active Operational Dataset from the response directly instead of decoding and encoding it again. Decoding the Active Operational Dataset will filter out any TLVs not known by the Commissioner, while returning Active Operational Dataset binary keeps it unchanged and has better compatibility. Thoughts?

Reply:
It make sense to me to output raw bytes without dissecting it. 
The problem is that there should be no such field as `mRawTlvs` in in the ActiveOperationalDataset struct. 
Moreover, the logic itself is not **decodeing** so it should not be in the method `DecodeActiveOperationalDataset`. 

I would suggest to use a new method (e.g. GetRawActiveOperationalDataset(aResponse)) to get the raw bytes. 

Thoughts?

Reply:
Works for me.

Reply:
done as suggested, please review @simonlingoogle .